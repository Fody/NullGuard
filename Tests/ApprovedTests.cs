#if (NET472)
using VerifyXunit;
using Fody;
using Xunit;
using System.Linq;
using System.Threading.Tasks;

[UsesVerify]
public class ApprovedTests
{
    [Fact]
    public Task ClassWithBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithBadAttributes"));
    }

    [Fact]
    public Task ClassWithPrivateMethod()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task ClassWithPrivateMethodNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task GenericClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClass`1"));
    }

    [Fact]
    public Task GenericClassWithValueTypeConstraint()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithValueTypeConstraints`1"));
    }

    [Fact]
    public Task GenericClassWithReferenceTypeConstraints()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithReferenceTypeConstraints`1"));
    }

    [Fact]
    public Task Indexers()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "Indexers"));
    }

    [Fact]
    public Task InterfaceBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "InterfaceBadAttributes"));
    }

    [Fact]
    public Task SimpleClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SimpleClassNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SkipIXamlMetadataProvider()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "XamlMetadataProvider"));
    }

#if (DEBUG)
    [Fact]
    public Task SpecialClass_debug()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#else
    [Fact]
    public Task SpecialClass_release()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#endif

    [Fact]
    public Task PublicNestedInsideNonPublic()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "NonPublicWithNested"));
    }

    [Fact]
    public Task UnsafeClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "UnsafeClass"));
    }

    [Fact]
    public Task ClassWithImplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithImplicitInterface"));
    }

    [Fact]
    public Task ClassWithExplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithExplicitInterface"));
    }

    [Fact]
    public Task InfosList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Messages.Select(x=>x.Text));
    }

    [Fact]
    public Task WarnsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Warnings.Select(x=>x.Text));
    }

    [Fact]
    public Task ErrorsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Errors.Select(x=>x.Text));
    }
}

#endif