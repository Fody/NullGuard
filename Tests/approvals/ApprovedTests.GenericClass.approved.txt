.class public auto ansi beforefieldinit GenericClass`1<T>
       extends [mscorlib]System.Object
{
  .field private !T '<NonNullProperty>k__BackingField'
  .method public hidebysig specialname instance !T 
          get_NonNullProperty() cil managed
  {
    // Code size       49 (0x31)
    .maxstack  3
    IL_0000:  ldarg.0
    IL_0001:  ldfld      !0 class GenericClass`1<!T>::'<NonNullProperty>k__BackingField'
    IL_0006:  dup
    IL_0007:  box        !T
    IL_000c:  ldnull
    IL_000d:  ceq
    IL_000f:  ldc.i4.0
    IL_0010:  ceq
    IL_0012:  ldstr      "[NullGuard] Return value of property 'T GenericCla"
    + "ss`1::NonNullProperty()' is null."
    IL_0017:  call       void [System]System.Diagnostics.Debug::Assert(bool,
                                                                       string)
    IL_001c:  dup
    IL_001d:  box        !T
    IL_0022:  brtrue.s   IL_0030
    IL_0024:  pop
    IL_0025:  ldstr      "[NullGuard] Return value of property 'T GenericCla"
    + "ss`1::NonNullProperty()' is null."
    IL_002a:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_002f:  throw
    IL_0030:  ret
  } 
  .method public hidebysig specialname instance void 
          set_NonNullProperty(!T 'value') cil managed
  {
    // Code size       54 (0x36)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  box        !T
    IL_0006:  ldnull
    IL_0007:  ceq
    IL_0009:  ldc.i4.0
    IL_000a:  ceq
    IL_000c:  ldstr      "[NullGuard] Cannot set the value of property 'T Ge"
    + "nericClass`1::NonNullProperty()' to null."
    IL_0011:  call       void [System]System.Diagnostics.Debug::Assert(bool,
                                                                       string)
    IL_0016:  ldarg.1
    IL_0017:  box        !T
    IL_001c:  brtrue.s   IL_002e
    IL_001e:  ldstr      "value"
    IL_0023:  ldstr      "[NullGuard] Cannot set the value of property 'T Ge"
    + "nericClass`1::NonNullProperty()' to null."
    IL_0028:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_002d:  throw
    IL_002e:  ldarg.0
    IL_002f:  ldarg.1
    IL_0030:  stfld      !0 class GenericClass`1<!T>::'<NonNullProperty>k__BackingField'
    IL_0035:  ret
  } 
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } 
  .property instance !T NonNullProperty()
  {
  } 
}
